flamingo/services/clone.py

```diff
@@ -0,0 +1,41 @@
+from dataclasses import dataclass
+from typing import Dict, Optional, Tuple
+
+from gcp_pilot.datastore import DoesNotExist
+
+from models.app import App
+from models.environment import Environment
+
+
+@dataclass
+class AppClone:
+    app: App
+
+    def clone(self, env_id: str) -> Tuple[Optional[App], bool, Optional[Dict]]:
+        if not env_id:
+            return None, False, {"environment_id": "attribute is required"}
```

Comment:
Raising errors seems way easier to handle in the view, don't you think?
The trio as return value seems too cryptic.

Reply:
Indeed. Fixed!